tubular/dates.py

```diff
@@ -663,3 +665,152 @@ def transform(self, X):
         X[self.new_column_name] = lower_comparison & upper_comparison
 
         return X
+
+
+class DatetimeSinusoidCalculator(BaseTransformer):
+
+    """
+    Transformer to derive a feature in a dataframe by calculating the
+    sine or cosine of a datetime column in a given unit (e.g hour), with the option to scale
+    period of the sine or cosine to match the natural period of the unit (e.g. 24).
+
+    Parameters
+    ----------
+    columns : str or list
+        Columns to take the sine or cosine of. Must be a datetime[64] column.
+
+    method : str or list
+        Argument to specify which function is to be calculated. Accepted values are 'sin', 'cos' or a list containing both.
+
+    units : str
+        Which time unit the calculation is to be carried out on. Accepted values are 'year', 'month',
+        'day', 'hour', 'minute', 'second', 'microsecond'.
+
+    period : int or float, default = 2*np.pi
+        The period of the output in the units specified above. To leave the period of the sinusoid output as 2 pi, specify 2*np.pi (or leave as default)
+
+    Attributes
+    -----------
+    columns : str or list
+        Columns to take the sine or cosine of.
+
+    method : str
+        The function to be calculated; either sin, cos or a list containing both.
+
+    units : str
+        Which time unit the calculation is to be carried out on. Will take any of 'year', 'month',
+        'day', 'hour', 'minute', 'second', 'microsecond'.
+
+    period : str or float, default = 2*np.pi
+        The period of the output in the units specified above.
+
+    """
+
+    def __init__(
+        self,
+        columns: Union[str, List[str]],
+        method: Union[str, List[str]],
+        units: str,
+        period: Union[int, float] = 2 * np.pi,
+    ):
+
+        super().__init__(columns, copy=True)
+
+        if not isinstance(method, str) and not isinstance(method, list):
+            raise TypeError(
+                "{}: method must be a string or list but got {}".format(
+                    self.classname(), type(method)
+                )
+            )
+
+        if not isinstance(units, str):
+            raise TypeError(
+                "{}: units must be a string but got {}".format(
+                    self.classname(), type(units)
+                )
+            )
+
+        if (
+            (not isinstance(period, int))
+            and (not isinstance(period, float))
+            or isinstance(period, bool)
```

Comment:
Think the bool option here is left over from when I had the option to set `period=False`

Reply:
removed this line and relevant test case: 
https://github.com/lvgig/tubular/pull/50/commits/eff8469d47e048a8634eb8227f5491c0a048c531